framework/src/utils/SolutionInvalidity.C

```diff
@@ -0,0 +1,147 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SolutionInvalidity.h"
+
+// MOOSE Includes
+#include "MooseError.h"
+#include "MooseApp.h"
+#include "VariadicTable.h"
+
+// System Includes
+#include <chrono>
+#include <memory>
+
+SolutionInvalidity::SolutionInvalidity(MooseApp & app)
+  : ConsoleStreamInterface(app),
+    _solution_invalidity_registry(moose::internal::getSolutionInvalidityRegistry())
+{
+}
+
+SolutionInvalidity::~SolutionInvalidity() {}
+/// Count solution invalid occurrences for each solution id
```

Comment:
All of the comments in these methods should be defined in the header file, not here